yaml/_loader/loader.ts

```diff
@@ -1545,14 +1547,16 @@ function composeNode(
         );
       }
 
-      if (!type.resolve(state.result)) {
+      if (!type.resolve?.(state.result) ?? true) {
```

Comment:
This condition is reversed when `resolve` is a default implementation

Reply:
Nice catch! Yeah, if we're to go through with this PR, let's make `Type.resolve()` and `Type.construct()` required.

Reply:
Done.